bundles/org.openhab.binding.wemo/src/main/resources/ESH-INF/thing/thing-types.xml

```diff
@@ -354,4 +380,50 @@
 		<description>Date/time the coffee maker last completed cleaning</description>
 	</channel-type>
 
+    <channel-type id="faderCountDownTime" advanced="true">

```

Comment:
Just a suggestion: Wouldn't it make more sense to consider the fading to be a configuration of the `brightness` channel? After all, it effects how this channel behaves, right?

Reply:
No, it is an independent feature of the device

Reply:
Could you explain this in some more detail? What is the dimmer actively doing when sending a command to `faderCountDownTime`?

Reply:
sending a command to faderCountDownTime will set the time the device will take to fade out. Like in original Wemo App, one can see the remaining time if it is actually fading.

Reply:
> will set the time

That's why I say that it is a configuration and not an action by itself. What triggers the fading? Sending a new value to `brightness`?

Reply:
No, it is triggered by sending a specific command to the device. You can switch fading on/off while device is physically on and no brightness is changed. As in my last comment, chnaging brightness does not switch the device.

Reply:
> it is triggered by sending a specific command to the device.

Let me be more precise: Which openHAB channel is used to trigger this specific command?

Reply:
Its is CHANNEL_TIMERSTART , Lines 204ff

Reply:
So isn't faderCountDownTime a configuration of CHANNEL_TIMERSTART then?

Reply:
If so, how would one present a channel configuration in the UI ?

Reply:
Presentation is done automatically by the UI, it's a standard feature, see e.g. LIFX:

<img width="1178" alt="Bildschirmfoto 2019-11-03 um 13 08 33" src="https://user-images.githubusercontent.com/3244965/68084821-22b00f80-fe3b-11e9-9b0f-a68715ba5776.png">

Reply:
I did not mean PaperUI, I was talking about BasicUI, ClassicUI or even habPanel. Do you really want the user to go to the Thing config when he wants to change the faderTime or the nightModeBrigthness ?

Reply:
Well, that's the idea for anything that concerns the system configuration. And aren't those values something that defines the configuration of your system setup (and thus is done by an "admin user"), while the rest of the users don't have to care about those (and rather should not touch them)?
I agree that we don't have very smooth admin UIs yet, but I'd hope that gets better with the new UI soon.
How is it implemented in the WeMo app? Do they show sliders for nightModeBrightness next to the "normal" brightness? Or isn't it maybe also in some config/admin parts of the app?

Reply:
No, It is not shown in a config part of the app. When you enter the nightMode Menu, you can see the slider just below the switch.

Reply:
> It is not shown in a config part of the app

Hm, "Nachtmoduseinstellungen" translates for me to just that: config settings. The page even has a "save" button at the top after reconfiguring the values.

Reply:
The save button sends the brightness and time settings to the device. Do you really aske me to change the code to read these settings as config parameters and send them to the device before switching nightmode on ? From user perspective, one has to open PaperUI, change the channel config in the thing settings cause there is no easy way to do this in the other UIs ?
Is this really what you want ? Not very user friendly from my perspective.

Reply:
Oops, sorry for not answering any quicker :-(.
I think what we are actually lacking is a solution for https://github.com/eclipse/smarthome/issues/3484. The features we talk about here are clearly a "configuration of the device", while not being a configuration of the handler. I agree with you that we do not have a good mechanism to sync those values (and their changes) between openHAB and the device, if they are modelled as config parameters. So in this light, let's be pragmatic and go the way you suggest and have them made available as (advanced) channels. For the users, it's indeed currently the most convenient option.

Reply:
Ok, thanks. If there would be a more user friendly way, I would take it and implement this as device config parameters.